ttnn/cpp/ttnn/operations/experimental/ccl/reduce_scatter_minimal_async/device/reduce_scatter_minimal_async_program.cpp

```diff
@@ -376,4 +432,371 @@ tt::tt_metal::operation::ProgramWithCallbacks reduce_scatter_minimal_async_helpe
+
+    // L1 Scratch CB Creation
+    const size_t packet_size_bytes = tt::tt_fabric::get_tt_fabric_channel_buffer_size_bytes();
+    uint32_t l1_scratch_cb_page_size_bytes = op_config.get_page_size();
+    uint32_t num_pages_per_packet = packet_size_bytes / l1_scratch_cb_page_size_bytes;
+    const uint32_t max_scatter_write_pages = 2;
+    const uint32_t max_dst_size = 8;  // TODO: generalize based on arch and fp32 acc
+    uint32_t tiles_to_write_per_packet = std::min(num_pages_per_packet, max_scatter_write_pages);
+    uint32_t tile_granularity = std::min(4 * num_pages_per_packet, max_dst_size);
+    uint32_t cb_num_pages = 3 * tile_granularity;  // triple buffering
+    tt::DataFormat df = tt::tt_metal::datatype_to_dataformat_converter(input_tensor.get_dtype());
+
+    uint32_t input_cb_index = tt::CB::c_in0;
+    tt::tt_metal::CircularBufferConfig cb_input_config =
+        tt::tt_metal::CircularBufferConfig(cb_num_pages * l1_scratch_cb_page_size_bytes, {{input_cb_index, df}})
+            .set_page_size(input_cb_index, l1_scratch_cb_page_size_bytes);
+    tt::tt_metal::CBHandle cb_input_workers = CreateCircularBuffer(program, sender_worker_core_range, cb_input_config);
+    uint32_t intermediate_cb_index = tt::CB::c_in1;
+    tt::tt_metal::CircularBufferConfig cb_intermediate_config =
+        tt::tt_metal::CircularBufferConfig(cb_num_pages * l1_scratch_cb_page_size_bytes, {{intermediate_cb_index, df}})
+            .set_page_size(intermediate_cb_index, l1_scratch_cb_page_size_bytes);
+    tt::tt_metal::CBHandle cb_intermediate_workers =
+        CreateCircularBuffer(program, sender_worker_core_range, cb_intermediate_config);
+    uint32_t reader_output_cb_index = tt::CB::c_in2;
+    tt::tt_metal::CircularBufferConfig cb_reader_output_config =
+        tt::tt_metal::CircularBufferConfig(cb_num_pages * l1_scratch_cb_page_size_bytes, {{reader_output_cb_index, df}})
+            .set_page_size(reader_output_cb_index, l1_scratch_cb_page_size_bytes);
+    tt::tt_metal::CBHandle cb_reader_output_workers =
+        CreateCircularBuffer(program, sender_worker_core_range, cb_reader_output_config);
+    uint32_t compute_output_cb_index = tt::CB::c_in3;
+    tt::tt_metal::CircularBufferConfig cb_compute_output_config =
+        tt::tt_metal::CircularBufferConfig(
+            cb_num_pages * l1_scratch_cb_page_size_bytes, {{compute_output_cb_index, df}})
+            .set_page_size(compute_output_cb_index, l1_scratch_cb_page_size_bytes);
+    tt::tt_metal::CBHandle cb_compute_output_workers =
+        CreateCircularBuffer(program, sender_worker_core_range, cb_compute_output_config);
+
+    // Set aside a buffer we can use for storing packet headers in (particularly for atomic incs)
+    const auto reserved_packet_header_CB_index = tt::CB::c_in4;
+    static constexpr auto num_packet_headers_storable = 4;
+    static constexpr auto packet_header_size_bytes = sizeof(tt::tt_fabric::PacketHeader);
```

Comment:
instead of sizeof(), please use the official API (I'm slowly removing this TT-Metal API violation from existing
```suggestion
    static constexpr auto packet_header_size_bytes = tt::tt_fabric::get_tt_fabric_packet_header_size_bytes();
```